examples/pytorch/rgcn/entity.py

```diff
@@ -1,71 +1,99 @@
 
-from entity_utils import load_data
-from model import RGCN
-
-def main(args):
-    g, num_rels, num_classes, labels, train_idx, test_idx, target_idx = load_data(
-        args.dataset, get_norm=True)
-
-    model = RGCN(g.num_nodes(),
-                 args.n_hidden,
-                 num_classes,
-                 num_rels,
-                 num_bases=args.n_bases)
-
-    if args.gpu >= 0 and th.cuda.is_available():
-        device = th.device(args.gpu)
-    else:
-        device = th.device('cpu')
-    labels = labels.to(device)
-    model = model.to(device)
-    g = g.int().to(device)
+class RGCN(nn.Module):
+    def __init__(self, num_nodes, h_dim, out_dim, num_rels):
+        super().__init__()
+        self.emb = nn.Embedding(num_nodes, h_dim)
+        # two-layer RGCN
+        self.conv1 = RelGraphConv(h_dim, h_dim, num_rels, regularizer='basis',
+                                  num_bases=num_rels, self_loop=False)
+        self.conv2 = RelGraphConv(h_dim, out_dim, num_rels, regularizer='basis',
+                                  num_bases=num_rels, self_loop=False)
+        
+    def forward(self, g):
+        x = self.emb.weight
+        h = F.relu(self.conv1(g, x, g.edata[dgl.ETYPE], g.edata['norm']))
+        h = self.conv2(g, h, g.edata[dgl.ETYPE], g.edata['norm'])
+        return h
+    
+def evaluate(g, target_idx, labels, test_mask, model):
+    test_idx = torch.nonzero(test_mask, as_tuple=False).squeeze()
+    model.eval()
+    with torch.no_grad():
+        logits = model(g)
+    logits = logits[target_idx]
+    return accuracy(logits[test_idx].argmax(dim=1), labels[test_idx]).item()
 
-    optimizer = th.optim.Adam(model.parameters(), lr=1e-2, weight_decay=args.wd)
+def train(g, target_idx, labels, train_mask, model):
+    # define train idx, loss function and optimizer
+    train_idx = torch.nonzero(train_mask, as_tuple=False).squeeze()
+    loss_fcn = nn.CrossEntropyLoss()
+    optimizer = torch.optim.Adam(model.parameters(), lr=1e-2, weight_decay=5e-4)
 
     model.train()
-    for epoch in range(100):
+    for epoch in range(50):
         logits = model(g)
         logits = logits[target_idx]
-        loss = F.cross_entropy(logits[train_idx], labels[train_idx])
+        loss = loss_fcn(logits[train_idx], labels[train_idx])
         optimizer.zero_grad()
         loss.backward()
         optimizer.step()
-
-        train_acc = accuracy(logits[train_idx].argmax(dim=1), labels[train_idx]).item()
-        print("Epoch {:05d} | Train Accuracy: {:.4f} | Train Loss: {:.4f}".format(
-            epoch, train_acc, loss.item()))
-    print()
-
-    model.eval()
-    with th.no_grad():
-        logits = model(g)
-    logits = logits[target_idx]
-    test_acc = accuracy(logits[test_idx].argmax(dim=1), labels[test_idx]).item()
-    print("Test Accuracy: {:.4f}".format(test_acc))
-
+        acc = accuracy(logits[train_idx].argmax(dim=1), labels[train_idx]).item()
+        print("Epoch {:05d} | Loss {:.4f} | Accuracy {:.4f} "
```

Comment:
"Accuracy" -> "Train Accuracy" to avoid potential confusion with things like "Val Accuracy".

Reply:
Thanks. Fixed.